ext/hook/uhook.c

```diff
@@ -250,6 +258,37 @@ static bool dd_uhook_begin(zend_ulong invocation, zend_execute_data *execute_dat
     }
     dyn->hook_data->execute_data = NULL;
 
+    if (dyn->hook_data->suppress_call) {
+        if (ZEND_USER_CODE(execute_data->func->type)) {
+            static union {
+                zend_op zop;
+                zval zv;
+            } retop[] = {[0].zop =
+                             {
+                                 .opcode = ZEND_RETURN,
+                                 .op1_type = IS_CONST,
+                                 .op1 = {.constant = sizeof(retop[0])},
+                                 .op2_type = IS_UNUSED,
+                             },
+                         [1].zv = {.u1.type_info = IS_NULL}};
+            // the race condition doesn't matter
+            if (!retop[0].zop.handler) {
+                zend_vm_set_opcode_handler(&retop[0].zop);
+            }
+            struct {
+                zend_function new_func;
+                zend_function *orig_func; } *fs = emalloc(sizeof *fs);
+            memcpy(&fs->new_func, execute_data->func, sizeof fs->new_func);
+            fs->orig_func = execute_data->func;
+            fs->new_func.op_array.last = 1;
+            fs->new_func.op_array.opcodes = (zend_op *)retop;
+            execute_data->func = &fs->new_func;
+            execute_data->opline = &retop[0].zop;
+        } else {
+            // XXX: not supported yet
```

Comment:
This one is going to be interesting to support at all... Because observers are called after the func is fetched in ZEND_DO_ICALL...

Reply:
In fact you probably need to temporarily replace `internal_function.handler` with your own function, which then will replace `internal_function.handler` with the original function again and call it. ... with a global refcount to avoid replacing it while another thread also has replaced it...